tests/swap_test.go

```diff
@@ -224,27 +222,26 @@ var _ = Describe("[Serial][sig-compute]SwapTest", Serial, decorators.SigCompute,
 		})
 
 		It("Migration of vmi to memory overcommited node", func() {
-			nodes := libnode.GetAllSchedulableNodes(virtClient).Items
-			targetNode := nodes[0]
-			sourceNode := nodes[1]
-
+			sourceNode, targetNode, err := getValidSourceNodeAndTargetNodeForHostModelMigration(virtClient)
+			Expect(err).ToNot(HaveOccurred(), "should be able to get valid source and target nodes for migartion")
 			vmMemoryRequestkib := 512000
-			availableMemSizeKib := getAvailableMemSizeInKib(targetNode)
-			availableSwapSizeKib := getSwapFreeSizeInKib(targetNode)
+			availableMemSizeKib := getAvailableMemSizeInKib(*targetNode)
+			availableSwapSizeKib := getSwapFreeSizeInKib(*targetNode)
 			swapSizeToUsekib := min(maxSwapSizeToUseKib, int(swapPartToUse*float64(availableSwapSizeKib)))
 
 			//make sure that the vm migrate data to swap space (leave enough space for the vm that we will migrate)
 			memToUseInTargetNodeVmKib := availableMemSizeKib + swapSizeToUsekib - vmMemoryRequestkib
 
-			//add label the node so we could schedule the vmi to it through the targert node selector
-			libnode.AddLabelToNode(targetNode.Name, "swaptest", "swaptest")
-			defer libnode.RemoveLabelFromNode(targetNode.Name, "swaptest")
-
 			//The vmi should have more memory than memToUseInTheVm
 			vmiMemSize := resource.MustParse(fmt.Sprintf("%dMi", int((float64(memToUseInTargetNodeVmKib)+float64(gigbytesInkib*2))/bytesInKib)))
 			vmiMemReq := resource.MustParse(fmt.Sprintf("%dMi", vmMemoryRequestkib/bytesInKib))
 			vmiToFillTargetNodeMem := tests.NewRandomFedoraVMIWithGuestAgent()
-			vmiToFillTargetNodeMem.Spec.NodeSelector = map[string]string{"swaptest": "swaptest"}
+			//we want vmiToFillTargetNodeMem to land on the target node to achieve memory-overcommitment in target
+			nodeAffinityRuleForVmiToFill, err := getAffinityForTargetNode(targetNode)
+			Expect(err).ToNot(HaveOccurred())
+			vmiToFillTargetNodeMem.Spec.Affinity = &v1.Affinity{
+				NodeAffinity: nodeAffinityRuleForVmiToFill,
+			}
```

Comment:
Nit: maybe `getAffinityForTargetNode()` could return a `*v1.Affinity` so its return value could be used directly?

Reply:
Done